packages/extension-ui/src/Popup/Signing/Extrinsic/Extrinsic.tsx

```diff
@@ -0,0 +1,220 @@
+        <div className='separator'></div>
+        <td className='from'>{url}</td>
+      </tr>
+      <tr>
+        <td className='label'>
+          {t<string>('nonce')}&nbsp;
+          <ExtrinsicTooltip content='The overall, lifetime transaction count of your account.'>
+            <Svg
+              className='help-icon'
+              src={helpIcon}
+            />
+          </ExtrinsicTooltip>
+        </td>
+        <div className='separator'></div>
+        <td className='data'>{formatNumber(Number(requestPayload.nonce))}</td>
+      </tr>
+      {transactionDetails && (Array.isArray(transactionDetails) ?
+        transactionDetails.map((batchTransactionDetails, i) => (
+          <>
+            <BatchRow key={i}>
+              <HeaderCell className='label'>{t<string>('Batched transaction')} {i + 1}/{transactionDetails.length}</HeaderCell>
+            </BatchRow>
+            <CallDefinitionRows {...batchTransactionDetails} />
+          </>
+        )) :
+        <CallDefinitionRows {...transactionDetails} />
+      )}
+    </FullWidthTable>
+  );
+}
+
+export default Extrinsic;
+
+const EllipsisCell = styled.td`
+  &&& { /* Override the intrusive <Table> styles */
+    display: block;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+`;
+
+const FullWidthTable = styled(Table)`
+  width: 100%;
+`;
+
+const HeaderCell = styled.td.attrs({ rowSpan: 3 })`
+  &&& { /* Override the intrusive <Table> styles */
+    color: #fff;
+    font-size: 16px;
+    white-space: nowrap;
+    font-weight: 500;
+    font-family: ${({theme}) => theme.secondaryFontFamily};
+    line-height: 125%;
+    letter-spacing: 0.96px;
+  }
+`;
+
+const SubHeaderCell = styled(HeaderCell)`
+  &&& { /* Override the intrusive <Table> styles */
+    font-size: 14px;
+    line-height: 120%;
+    letter-spacing: 0.98px;
+    margin-top: 10px;
+  }
+`;
+
+const BatchRow = styled.tr`
+  margin-top: 40px;
+`;
+
+const CallDefinitionRows = (props: { section: string, method: string, args: { name: string, value?: Renderable }[], docs?: string[]}) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <tr>
+        <td className='label'>{t<string>('Section')}</td>
+        <div className='separator'></div>
+        <EllipsisCell className='data'>{props.section}</EllipsisCell>
+      </tr>
+      <tr>
+        <td className='label'>{t<string>('Method')}</td>
+        <div className='separator'></div>
+        <EllipsisCell className='data'>{props.method}</EllipsisCell>
+      </tr>
+      {props.args?.length && (
+        <tr>
+          <SubHeaderCell className='label'>{t<string>('Arguments')}</SubHeaderCell>
+        </tr>
+      )}
+      {props.args.map((arg) => (
+        <tr key={arg.name}>
+          <td className='label'>{arg.name}</td>
+          <div className='separator'></div>
+          <EllipsisCell className='data'>
+            <ArgumentValue>{arg.value}</ArgumentValue>
```

Comment:
those are all same account, but because of a shift I got confused at first, not sure if there is anything we can do :shrug: 
![image](https://github.com/Cardinal-Cryptography/aleph-zero-signer/assets/35614528/bf5b35f7-657a-4513-866c-cac062549652)

Reply:
Welp, I made it so that they take up the available space (modulo some margin). The alternative is to make the chars cound static (which is how it used to be implemented), but then we face 2 advantages: less charaters displayed than might have been and sometimes not enough space to fit the entire string.